pcsx2/gui/MemoryCardFolder.cpp

```diff
@@ -450,17 +445,21 @@ bool FolderMemoryCard::AddFile( MemoryCardFileEntry* const dirEntry, const wxStr
 		}
 
 		newFileEntry->entry.data.length = filesize;
-		u32 fileDataStartingCluster = GetFreeDataCluster();
-		newFileEntry->entry.data.cluster = fileDataStartingCluster;
-
-		// mark the appropriate amount of clusters as used
-		u32 dataCluster = fileDataStartingCluster;
-		m_fat.data[0][0][dataCluster] = LastDataCluster | DataClusterInUseMask;
-		for ( unsigned int i = 0; i < countClusters - 1; ++i ) {
-			u32 newCluster = GetFreeDataCluster();
-			m_fat.data[0][0][dataCluster] = newCluster | DataClusterInUseMask;
-			m_fat.data[0][0][newCluster] = LastDataCluster | DataClusterInUseMask;
-			dataCluster = newCluster;
+		if ( filesize != 0 ) {
+			u32 fileDataStartingCluster = GetFreeDataCluster();
+			newFileEntry->entry.data.cluster = fileDataStartingCluster;
+
+			// mark the appropriate amount of clusters as used
+			u32 dataCluster = fileDataStartingCluster;
+			m_fat.data[0][0][dataCluster] = LastDataCluster | DataClusterInUseMask;
+			for ( unsigned int i = 0; i < countClusters - 1; ++i ) {
+				u32 newCluster = GetFreeDataCluster();
+				m_fat.data[0][0][dataCluster] = newCluster | DataClusterInUseMask;
+				m_fat.data[0][0][newCluster] = LastDataCluster | DataClusterInUseMask;
+				dataCluster = newCluster;
+			}
+		} else {
+			newFileEntry->entry.data.cluster = MemoryCardFileEntry::EmptyFileCluster;
```

Comment:
This seems a but weird.

An empty file is valid on most file systems. Wouldn't it be better to setup a valid empty file entry? i.e. with `newFileEntry->entry.data.length = 0` and the rest of the properties consistent with that? Using a magic number for the cluster in order to mark an empty file is iffy IMO.